Update heroku/java to 0.3.9 and heroku/java-function to 0.3.14

edmorley opened the pull request:
## `heroku/java`
### [0.3.9] 2021/07/28
* Upgraded `heroku/jvm` to `0.1.7`

## `heroku/java-function`
### [0.3.14] 2021/07/28
* Upgraded `heroku/jvm-function-invoker` to `0.4.0`
* Upgraded `heroku/jvm` to `0.1.7`

### [0.3.13] 2021/07/16
* Upgraded `heroku/maven` to `0.2.4`
* Upgraded `heroku/jvm-function-invoker` to `0.3.0`

---

The above changes bring in these additional changes...

## `heroku/jvm`
### [0.1.7] 2021/07/28
* Default version for **OpenJDK 7** is now `1.7.0_312`
* Default version for **OpenJDK 8** is now `1.8.0_302`
* Default version for **OpenJDK 11** is now `11.0.12`
* Default version for **OpenJDK 13** is now `13.0.8`
* Default version for **OpenJDK 15** is now `15.0.4`
* Default version for **OpenJDK 16** is now `16.0.2`

## `heroku/jvm-function-invoker`
### [0.4.0] 2021/07/28
* Revert all changes from `0.3.0`
* Updated function runtime to `1.0.0` again

### [0.3.0] 2021/07/15
* Changed implementation to Rust
* Updated function runtime to `1.0.0`

## `heroku/maven`
## [0.2.4] 2021/07/16
### Added
* Loosen stack requirements allowing any linux distro use this buildpack

Closes GUS-W-9630536.

Malax reviewed (commented):
`heroku/maven` and `heroku/jvm` are referenced directly in the builders as well. I always updated those to point to the new versions as well. Do we still want to do that? If not, why do we have them in the builder anyways? I don't know enough about builders to have a strong opinion here.

edmorley commented:
Ah I hadn't seen those. They are redundant - I've opened #188 to remove them and will merge that first, then rebase this.

edmorley commented:
I've rebased, then also updated the buildpack URLs to use the `heroku-buildpacks` ECR alias, to save having to wait until the next round of releases to pick up heroku/buildpacks-jvm/pull/128.

edmorley commented:
Functions canary run:
https://app.circleci.com/pipelines/github/heroku/function-cli-canary/443/workflows/d1d0494f-df58-4c46-b5cf-a142b37c2820/jobs/517
